packages/nx/src/native/pseudo_terminal/process_killer/unix.rs

```diff
@@ -0,0 +1,59 @@
+use nix::{
+    sys::signal::{kill, Signal as NixSignal},
+    unistd::Pid,
+};
+
+pub struct ProcessKiller {
+    pid: i32,
+}
+
+impl ProcessKiller {
+    pub fn new(pid: i32) -> Self {
+        Self { pid }
+    }
+
+    pub fn kill(&self, signal: Option<&str>) -> anyhow::Result<()> {
+        let pid = Pid::from_raw(self.pid);
+        match kill(
+            pid,
+            NixSignal::from(
+                Signal::try_from(signal.unwrap_or("SIGINT")).map_err(|e| anyhow::anyhow!(e))?,
+            ),
+        ) {
+            Ok(_) => Ok(()),
+            Err(e) => Err(anyhow::anyhow!("Failed to kill process: {}", e)),
+        }
+    }
+}
+
+enum Signal {
+    SIGTERM,
+    SIGINT,
+    SIGKILL,
+    SIGHUP,
+}
+
+impl TryFrom<&str> for Signal {
+    type Error = String;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        match value {
+            "SIGHUP" => Ok(Signal::SIGHUP),
+            "SIGINT" => Ok(Signal::SIGINT),
+            "SIGKILL" => Ok(Signal::SIGKILL),
+            "SIGTERM" => Ok(Signal::SIGTERM),
+            _ => Err(format!("Invalid signal: {}", value)),
+        }
+    }
+}
+
+impl From<Signal> for NixSignal {
```

Comment:
future: Is it possible to do `impl TryFrom<&str> for NixSignal` or in fact, is that already implemented?

Reply:
I tried that first, rust won't let you implement from one type to another unless at least one of the types is defined in the file